### PR TITLE
Update prose scope and basic styling

### PR DESCRIPTION
**Pull request by thisisdano**
This makes the global styling options and prose scope a bit clearer.

Removed `$theme-content-styles` and its options (`global`, `scoped`, `none`). Now content styles are _always_ applied to elements within `usa-prose` — this is the purpose of `usa-prose`.

Replaced the `$theme-global-styles-basic` bool with three explicit global styling bool vars. Each of these apply styling to unclassed elements when set to true:
- `$theme-global-paragraph-styles`
- `$theme-global-link-styles`
- `$theme-global-content-styles`

Each of these values is now set to `false` by default. We added explicit link styling to any of our components that depended on `basic-styles` being set to `true` to display link styles properly — and set overrides for any styles that would have been clobbered by global content styling. Now our components will display proper styling regardless of the value of these vars.

Now USWDS will only apply global styling when the user enables it.

`usa-prose` now has a single use: adding direct-child content styling within its context. We will add `usa-prose` documentation in a subsequent `uswds-site` PR. 

| previous setting | Beta 6 equivalent
| --- | ---
| `$theme-content-styles: 'scoped'` | _no explicit setting necessary_
| `$theme-content-styles: 'global'` | `$theme-global-content-styles:   true`
| `$theme-content-styles: 'none'` | `$theme-global-paragraph-styles:   false` 
| | `$theme-global-link-styles:   false`
| | `$theme-global-content-styles:   false`
| `$theme-global-styles-basic: true` | `$theme-global-paragraph-styles:   true` 
| | `$theme-global-link-styles:   true`

### CODE REVIEW

File: src/stylesheets/components/_alerts.scss [hunk 1]
```diff
@@ -100,6 +91,8 @@ $padding-left: units($theme-alert-padding-x) + units($theme-alert-bar-width);
 
 .usa-alert-text {
   @include u-margin-y(0);
+
+  a { @include typeset-link; }
```
Comment (maya) [inline src/stylesheets/components/_alerts.scss]: Can this be written as:
```
a { 
  @include typeset-link; 
}
```
to be consistent with the rest of the code base?

File: src/stylesheets/settings/_settings-typography.scss [hunk 2]
```diff
@@ -53,30 +53,21 @@ $theme-root-font-size: 10px !default;
...
-Content styles
-----------------------------------------
-Sets the default typography styles for
-paragraph text, links, headings, lists,
-and tables
-
-Accepts global, scoped, or none
+- paragraph: paragraph text
+- link: links
+- global: paragraph text, links,
```
Comment (maya) [inline src/stylesheets/settings/_settings-typography.scss]: aren't they all global? should it read "content" or "content styles" instead?

File: src/components/01-type/links.njk [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-<p><a class="usa-color-text-visited" href="javascript:void(0);">This</a> is a visited link.</p>
+<p><a class="usa-link usa-color-text-visited" href="javascript:void(0);">This</a> is a visited link.</p>
 
-<p>This is a link that goes to an <a class="usa-external_link" href="https://media.giphy.com/media/8sgNa77Dvj7tC/giphy.gif">external website</a>.</p>
+<p>This is a link that goes to an <a class="usa-link usa-external_link" href="https://media.giphy.com/media/8sgNa77Dvj7tC/giphy.gif">external website</a>.</p>
 
 <div class="usa-background-dark">
-  <p><a href="javascript:void(0);">This</a> is a text link on a dark background.</p>
+  <p><a class="usa-link" href="javascript:void(0);">This</a> is a text link on a dark background.</p>
 </div>
```
Comment (maya) [inline src/components/01-type/links.njk]: Is a prose scope example represented somewhere on the site/documentation?

**Comment by maya** [comment]
Looks like there are some merge conflicts. Can you resolve the conflicting files?

File: src/components/01-type/links.njk (hunk 3 above)
Comment (maya) [reply src/components/01-type/links.njk]: Looks like we do use it on [Typesetting](https://v2.designsystem.digital.gov/components/typography/), though the documentation needs updating.

File: src/stylesheets/settings/_settings-typography.scss (hunk 2 above)
Comment (thisisdano) [reply src/stylesheets/settings/_settings-typography.scss]: Yeah, this should say `content styles`

File: src/components/01-type/links.njk (hunk 3 above)
Comment (thisisdano) [reply src/components/01-type/links.njk]: Updating prose scope docs is next on my list

File: src/stylesheets/components/_alerts.scss (hunk 1 above)
Comment (thisisdano) [reply src/stylesheets/components/_alerts.scss]: yeah, will do

**Comment by thisisdano** [comment]
OK, non-docs fixes and merge conflicts are done

**Comment by maya** [comment]
@thisisdano in the PR description you write the current setting is `$theme-global-content-styles: true`, but in the settings file it's set to `false`. Can you clarify what the state should be?

**Comment by thisisdano** [comment]
Do you mean in the table? I think I was unclear — what I meant to be saying there was "If you had this setting in Beta 5, use this new setting in Beta 6". The default is `$theme-global-content-styles: false`

**Comment by maya** [comment]
Got it now, thanks 👍

**Pull request merged**